source/game.production.scripttemplate.bmx

```diff
@@ -49,32 +83,13 @@ Type TScriptTemplateCollection Extends TGameObjectCollection
 	End Method
 
 
-	Method GetRandomByFilter:TScriptTemplate(skipNotAvailable:int = True, skipEpisodes:int = True, containsKeywords:string="", avoidIDs:int[] = null)
+	Method GetRandomByFilter:TScriptTemplate(filter:SScriptTemplateFilter)
```

Comment:
Nur als Hinweis:
Wenn Du `Method GetRandomByFilter:TScriptTemplate(filter:SScriptTemplateFilter)` zu `Method GetRandomByFilter:TScriptTemplate(filter:SScriptTemplateFilter Var)` machen wuerdest, erzeugt man keine Kopie des Structs (was aber trotzdem auf dem Stack passiert - also relativ "preisguenstig").

Falls man den Filter "aufheben" will, boete sich evtl ein "Wrapper" an:
```
Type TScriptTemplateFilter
  Field f:SScriptFilter
End Type
```
(zusammen mit dem `var` und einem Overload 
```
Method GetRandomByFilter:TScriptTemplate(filter:TScriptTemplateFilter)
  Return GetRandomByFilter(filter.f)
End Method
```
Haette man dann denke ich alles "abgedeckt".


Aber - das ist wie gesagt nur ein Hinweis (zum "Dazulernen") und nichts, was Du unbedingt umsetzen solltest.

Reply:
Mit der Übergabe von Pointern in Blitzmax habe ich mich tatsächlich noch nie beschäftigt. Da in der Suchmethode selbst der Filter nicht manipuliert werden soll sondern höchstens in der aufrufenden Methode, wäre ein Ergänzen des var wohl komlpett unproblematisch. Für das Aufheben eines Filters sehe ich derzeit keinen Anwendungsfall.